test/core/memory_usage/callback_client.cc

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include <gtest/gtest.h>
+
+#include "absl/algorithm/container.h"
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "util/logging.h"
+
+#include <grpc/byte_buffer.h>
+#include <grpc/byte_buffer_reader.h>
+#include <grpc/grpc.h>
+#include <grpc/grpc_security.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include <grpc/support/time.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/support/client_callback.h>
+
+#include "src/core/lib/address_utils/sockaddr_utils.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/gpr/env.h"
+#include "src/core/lib/gpr/string.h"
+#include "src/core/lib/gpr/useful.h"
+#include "src/core/lib/gprpp/host_port.h"
+#include "src/core/lib/iomgr/exec_ctx.h"
+#include "src/core/lib/iomgr/sockaddr.h"
+#include "src/core/lib/iomgr/socket_utils.h"
+#include "src/core/lib/security/credentials/credentials.h"
+#include "src/cpp/client/secure_credentials.h"
+#include "src/proto/grpc/testing/benchmark_service.grpc.pb.h"
+#include "test/core/memory_usage/memstats.h"
+#include "test/core/util/port.h"
+#include "test/core/util/subprocess.h"
+#include "test/core/util/test_config.h"
+
+ABSL_FLAG(std::string, target, "", "Target host:port");
+ABSL_FLAG(bool, secure, false, "Use SSL Credentials");
+
+void UnaryCall() {
+  // Set the authentication mechanism.
+  std::shared_ptr<grpc::ChannelCredentials> creds =
+      grpc::InsecureChannelCredentials();
+  if (absl::GetFlag(FLAGS_secure)) {
+    // TODO (chennancy) Add in secure credentials
+    gpr_log(GPR_INFO, "Supposed to be secure, is not yet");
+  }
+
+  // Create a channel to the server and a stub
+  std::shared_ptr<grpc::Channel> channel =
+      CreateChannel(absl::GetFlag(FLAGS_target), creds);
+  std::unique_ptr<grpc::testing::BenchmarkService::Stub> stub =
+      grpc::testing::BenchmarkService::NewStub(channel);
+
+  // Start a call.
+  struct CallParams {
+    grpc::ClientContext context;
+    grpc::testing::SimpleRequest request;
+    grpc::testing::SimpleResponse response;
+  };
+  CallParams* params = new CallParams();
```

Comment:
Can you comment on the ownership of params (what deletes it and where)

Reply:
Hm IIRC, if we delete the params in the callback, the connection closes; where would be an appropriate place to delete them? Perhaps we remove the struct and avoid heap-allocated memory here?